util/jsonpull.c

```diff
@@ -0,0 +1,974 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+#include "jsonpull.h"
+
+#include <assert.h>
+
+#define GVM_JSON_CHAR_EOF -1       ///< End of file
+#define GVM_JSON_CHAR_ERROR -2     ///< Error reading file
+#define GVM_JSON_CHAR_UNDEFINED -3 ///< Undefined state
+
+/**
+ * @brief Escapes a string according to the JSON or JSONPath standard
+ */
+gchar *
+gvm_json_string_escape (const char *string, gboolean single_quote)
+{
+  gchar *point;
+  if (string == NULL)
+    return NULL;
+
+  GString *escaped = g_string_sized_new (strlen (string));
+  for (point = (char *) string; *point != 0; point++)
+    {
+      unsigned char character = *point;
+
+      if ((character > 31) && (character != '\\')
+          && (single_quote || character != '\"')
+          && (!single_quote || character != '\''))
```

Comment:
```suggestion
          && (single_quote ? (character != '\'') : (character != '\"')))
```
I would find this easier to understand.